<?xml version="1.0" encoding="UTF-8"?>
<!-- Generated by org.testng.reporters.JUnitReportReporter -->
<testsuite hostname="DESKTOP-VGN6VUA" failures="1" tests="4" name="tests.CustomerTest" time="44.007" errors="0" timestamp="2025-01-09T22:32:03 IST" skipped="0">
  <testcase classname="tests.CustomerTest" name="VerifyUserIsAbleToLogin" time="6.512"/>
  <system-out/>
  <testcase classname="tests.CustomerTest" name="VerifyUserIsAbleToAddCustomer" time="8.567"/>
  <system-out/>
  <testcase classname="tests.CustomerTest" name="Dashboard" time="2.778"/>
  <system-out/>
  <testcase classname="tests.CustomerTest" name="VerifyUserIsAbleToUpdateCustomer" time="26.150">
    <failure message="Expected condition failed: waiting for visibility of element located by By.xpath: //button[normalize-space()=&#039;Update Customer&#039;] (tried for 20 second(s) with 500 milliseconds interval)" type="java.lang.AssertionError">
      <![CDATA[java.lang.AssertionError: Expected condition failed: waiting for visibility of element located by By.xpath: //button[normalize-space()='Update Customer'] (tried for 20 second(s) with 500 milliseconds interval)
at org.testng.Assert.fail(Assert.java:110)
at web.pages.BaseMethods.waitForElementVisibility(BaseMethods.java:152)
at web.pages.BaseMethods.clickElement(BaseMethods.java:294)
at web.pages.CustomerPage.updateCustomer(CustomerPage.java:80)
at tests.CustomerTest.VerifyUserIsAbleToUpdateCustomer(CustomerTest.java:58)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
at org.testng.TestRunner.privateRun(TestRunner.java:808)
at org.testng.TestRunner.run(TestRunner.java:603)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
at org.testng.SuiteRunner.run(SuiteRunner.java:326)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
at org.testng.TestNG.runSuites(TestNG.java:1092)
at org.testng.TestNG.run(TestNG.java:1060)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
    </failure>
  </testcase> <!-- VerifyUserIsAbleToUpdateCustomer -->
  <system-out/>
</testsuite> <!-- tests.CustomerTest -->
